cmd/contractcmd/init_validator_manager.go

```diff
@@ -186,13 +181,18 @@ func initValidatorManager(_ *cobra.Command, args []string) error {
 		OwnerAddress:        &ownerAddress,
 		RPC:                 initValidatorManagerFlags.RPC,
 	}
-	err = signatureaggregator.CreateSignatureAggregatorInstance(app, subnetID.String(), network, extraAggregatorPeers, aggregatorLogger, "latest")
-	if err != nil {
-		return err
-	}
-	signatureAggregatorEndpoint, err := signatureaggregator.GetSignatureAggregatorEndpoint(app, network)
-	if err != nil {
-		return err
+	var signatureAggregatorEndpoint string
+	if initValidatorManagerFlags.SigAggFlags.SignatureAggregatorEndpoint == "" {
+		err = signatureaggregator.CreateSignatureAggregatorInstance(app, network, aggregatorLogger, "latest")
```

Comment:
you should first try to get the endpoint, and only create instance in case of error

Reply:
addressed